artifact/artifact.sh

```diff
@@ -1,48 +1,89 @@
 #!/bin/bash -eux
-#
-# This code is meant to be executed within a TravisCI build.
+
+# This code is meant to be executed within a CI build.
 
 publish_to_dockerhub() {
-    if [ -n "${DOCKER_USERNAME:-}" ] && [ -n "${DOCKER_PASSWORD:-}"  ]; then
-        DOCKER_REPO="tidepool/${TRAVIS_REPO_SLUG#*/}"
-        echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin
+    if [ -n "${DOCKER_USERNAME:-}" ] && [ -n "${DOCKER_PASSWORD:-}" ]; then
+
+        # Determine CI provider
+        if [ -n "${TRAVIS:-}" ]; then
+            CI_PROVIDER="travis"
+        elif [ -n "${CIRCLECI:-}" ]; then
+            CI_PROVIDER="circle"
+        else
+            echo "No known CI provider detected"
+            return 1
+        fi
+
+        # Set common variables
+        if [ "$CI_PROVIDER" = "travis" ]; then
+            REPO_SLUG=$TRAVIS_REPO_SLUG
+            COMMIT=$TRAVIS_COMMIT
+            TAG=${TRAVIS_TAG:-}
+            BRANCH=${TRAVIS_BRANCH:-}
+            PR_NUMBER=${TRAVIS_PULL_REQUEST:-}
+        elif [ "$CI_PROVIDER" = "circle" ]; then
+            REPO_SLUG="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+            COMMIT=$CIRCLE_SHA1
+            TAG=${CIRCLE_TAG:-}
+            BRANCH=${CIRCLE_BRANCH:-}
+            PR_NUMBER=${CIRCLE_PR_NUMBER:-}
+        fi
```

Comment:
```suggestion
        if [ "$CI_PROVIDER" = "travis" ]; then
            REPO_SLUG=${TRAVIS_REPO_SLUG:?}
            COMMIT=${TRAVIS_COMMIT:?}
            TAG=${TRAVIS_TAG:?}
            BRANCH=${TRAVIS_BRANCH:?}
            PR_NUMBER=${TRAVIS_PULL_REQUEST:-}
        elif [ "$CI_PROVIDER" = "circle" ]; then
            REPO_SLUG="${CIRCLE_PROJECT_USERNAME:?}/${CIRCLE_PROJECT_REPONAME:?}"
            COMMIT=$CIRCLE_SHA1
            TAG=${CIRCLE_TAG:?}
            BRANCH=${CIRCLE_BRANCH:?}
            PR_NUMBER=${CIRCLE_PR_NUMBER:-}
        fi
```

Reply:
Optional: This is yet-another bashism, but a super helpful one. The `${FOO:?}` syntax says that `$FOO` isn't set, or is empty, to print an error and exit.

From my reading of the code it seems like the PR is optional, but the branch, commit, repo, etc are not. Feel free to adjust if I've missed something.